src/Microsoft.ML.AutoML/API/ExperimentBase.cs

```diff
@@ -194,7 +194,8 @@ public ExperimentResult<TMetrics> Execute(IDataView trainData, IDataView validat
             IProgress<CrossValidationRunDetail<TMetrics>> progressHandler = null)
         {
             UserInputValidationUtil.ValidateNumberOfCVFoldsArg(numberOfCVFolds);
-            var splitResult = SplitUtil.CrossValSplit(Context, trainData, numberOfCVFolds, columnInformation?.SamplingKeyColumnName);
+            UserInputValidationUtil.ValidateSamplingKey(columnInformation?.SamplingKeyColumnName, columnInformation?.GroupIdColumnName, _task);
+            var splitResult = SplitUtil.CrossValSplit(Context, trainData, numberOfCVFolds, columnInformation?.GroupIdColumnName);
```

Comment:
**Potential Bug** With the implementation you have here, if the user sets `SamplingKeyColumnName` for a non-ranking task, we'd still be passing "columnInformation?.GroupIdColumnName" which would be wrong for non-ranking tasks.

Also, if the user doesn't provide a `columnInformation` object, (i.e. `columnInformation = null`), then we would set `samplingKeyColumn = null` when splitting the data, which is wrong for Ranking tasks. Users should always provide a `GroupIdColumn`.